code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,265 @@
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return ITEM_INTERACT_BLOCKING
+	qdel(weapon)
+	core_installed = TRUE
+	update_anomaly_state()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/crate/crate_open.ogg', 50, FALSE)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/clothing/head/helmet/perceptomatrix/functioning
+	core_installed = TRUE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination
+	name = "Hallucinate"
+	desc = "Redirect perceptual energies towards a target, staggering them."
+	button_icon_state = "blind"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/blind_target.dmi'
+
+	sound = 'sound/items/weapons/emitter2.ogg'
+	school = SCHOOL_PSYCHIC
+	cooldown_time = 15 SECONDS
+
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+
+	active_msg = "You prepare to zap a target with hallucinations..."
+
+	/// The amount of blurriness to apply
+	var/eye_blur_duration = 7 SECONDS
+	/// The amount of stagger to apply
+	var/stagger_duration = 3 SECONDS
+	/// The amount of hallucination to apply
+	var/hallucination_duration = 25 SECONDS
+	/// Helmet the spell's bound to.
+	var/obj/item/clothing/head/helmet/perceptomatrix/linked_helmet
+	/// Spark system
+	var/datum/effect_system/spark_spread/quantum/spark_sys
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/New(Target)
+	. = ..()
+
+	linked_helmet = target
+	spark_sys = new /datum/effect_system/spark_spread/quantum
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/Destroy()
+	linked_helmet = null
+	QDEL_NULL(spark_sys)
+	return ..()
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/is_valid_target(atom/cast_on)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(ishuman(cast_on))
+		return TRUE
+	if(istype(cast_on, /obj/item/food/pancakes))
+		return TRUE
+
+	return FALSE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/can_cast_spell(feedback = TRUE)
+	. = ..()
+	if(!.)
+		return .
+
+	if(!linked_helmet)
+		stack_trace("casting w/o linked perceptomatrix!")
+
+	return .
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/proc/blows_up_pancakes_with_mind(obj/item/food/pancakes/pancakes)
+	cast_fx(pancakes)
+
+
+	owner.visible_message(
+		span_userdanger("[owner] blows up [pancakes] with [owner.p_their()] mind!"),
+		span_userdanger("You blow up [pancakes] with your mind!")
+	)
+
+	for(var/mob/chef in get_hearers_in_view(7, pancakes))
+		if(!chef.mind)
+			continue
+		if(!HAS_TRAIT_FROM(pancakes, TRAIT_FOOD_CHEF_MADE, REF(chef.mind)))
+			continue
+		if(prob(5) || check_holidays(APRIL_FOOLS))
+			chef.say("Ma fuckin' pancakes!")
+			playsound(pancakes, 'sound/effects/fuse.ogg', 80)
+			animate(pancakes, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
+			animate(pancakes, time = 1, pixel_z = 0, easing = BOUNCE_EASING)
+			for(var/i in 1 to 15)
+				animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING)
```

Comment:
Is this animate working as it should in testing? 

From past experience this isn't how you queue animates in a loop so I would be surprised if it functions

(Disregard this if it actually works I guess)

```suggestion
			animate(pancakes, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
			animate(time = 1, pixel_z = 0, easing = BOUNCE_EASING)
			for(var/i in 1 to 15)
				animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING, flags = ANIMATION_CONTINUE)
```